vispy/scene/visuals.py

```diff
@@ -13,11 +13,88 @@
 `vispy.visuals.Visual` rather than `vispy.scene.Node`.
 """
 import re
+import weakref
 
 from .. import visuals
 from .node import Node
+from ..visuals.filters import ColorFilter, PickingFilter
 
 
+class VisualNode(Node):
+    _next_id = 1
+    _visual_ids = weakref.WeakValueDictionary()
+    
+    def __init__(self, parent=None, name=None):
+        Node.__init__(self, parent=parent, name=name,
+                      transforms=self.transforms)
+        self.interactive = False
+        self._opacity_filter = ColorFilter()
+        self.attach(self._opacity_filter)
+        
+        self._id = VisualNode._next_id
+        VisualNode._visual_ids[self._id] = self
+        VisualNode._next_id += 1
+        self._picking_filter = PickingFilter(id_=self._id)
+        self.attach(self._picking_filter)
+
+    def _update_opacity(self):
+        self._opacity_filter.color = (1, 1, 1, self._opacity)
+        
+    def _set_clipper(self, node, clipper):
+        """Assign a clipper that is inherited from a parent node.
+        
+        If *clipper* is None, then remove any clippers for *node*.
+        """
+        if node in self._clippers:
+            self.detach(self._clippers.pop(node))
+        if clipper is not None:
+            self.attach(clipper)
+            self._clippers[node] = clipper
+
+    @property
+    def picking(self):
+        """Boolean that determines whether this node (and its children) are
+        drawn in picking mode.
+        """
+        return self._picking
```

Comment:
It is weird. I wend back and forth on it a few times. I'd be happy for it to change in the future if there is good cause. If the user sets it True, then the object will be rendered using its solid picking ID color.